dashboard/app/models/user.rb

```diff
@@ -958,6 +958,27 @@ def student_of?(teacher)
     teachers.include? teacher
   end
 
+  # @return {boolean} true if (1) Attended CSD TeacherCon '17 (2) are a CSD facilitator
+  def circuit_playground_pd_eligible?
+    csd_cohorts = %w(CSD-TeacherConPhiladelphia CSD-TeacherConPhoenix CSD-TeacherConHouston)
+
+    return true if cohorts.any? {|cohort| csd_cohorts.include?(cohort.name)}
+    return true if courses_as_facilitator.any? {|course_facilitator| course_facilitator.course == Pd::Workshop::COURSE_CSD}
+    return false
+  end
+
+  # @return {boolean} true if we have at least one section that meets our eligibility
+  #   requirements for student progress
+  def circuit_playground_student_progress_eligible?
+    sections.any?(&:has_sufficient_discount_code_progress?)
```

Comment:
I think this check should also check across all StudioPerson linked accounts - right now, you require the teacher to be signed in under the account owning the sections with sufficient progress.

Reply:
This is the intention. We check across all StudioPerson accounts for PD, but you need to have progress with your signed in account.